### PR TITLE
Add basic CI testing as Github actions

### PR DESCRIPTION
**Pull request by keithbennett**

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    strategy:
+      matrix:
+        include:
+        - mode: debug
+          flag: -b
+          define:
+        - mode: debug
+          flag: -b
+          define: >-
+            -DPER_SPECIES_WEIGHT -DPARTICLE_ID -DPHOTONS -DTRIDENT_PHOTONS
```
Comment (TomGoffrey) [inline .github/workflows/build_test.yml:26]: What's the motivation behind this list? Is there a reason we don't include bremsstrahlung for example? What about non-default particle weighting?

Comment (keithbennett) [reply .github/workflows/build_test.yml:26]: These are just all the compile-time flags that aren't set by default. They are all enabled here to check that all of those ifdef'ed sections actually compile. No doubt there are still unchecked parts of the code, but this hopefully reduces them. What things we actually choose to enable by default is a whole separate discussion.

Comment (keithbennett) [reply .github/workflows/build_test.yml:26]: I thought that one of the non-default weightings was included. Looks like I missed that. Will add it.

Comment (keithbennett) [reply .github/workflows/build_test.yml:26]: Also missed bremsstrahlung

Comment (keithbennett) [reply .github/workflows/build_test.yml]: Done now.

**Review by TomGoffrey — APPROVED** [review]

**Pull request merged**